SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_BlueberryQuests.h

```diff
@@ -0,0 +1,117 @@
+    up,
+    down
+};
+
+//Basic blue quests
+const std::set<BBQuests> blue_quests = {
+    BBQuests::auto_10, BBQuests::make_tm, BBQuests::pickup_10, BBQuests::sneak_up, BBQuests::photo_fly, BBQuests::photo_swim, BBQuests::photo_canyon, BBQuests::photo_coastal, BBQuests::photo_polar, BBQuests::photo_savanna, BBQuests::tera_self_defeat, 
+    BBQuests::travel_500, BBQuests::catch_any, BBQuests::catch_normal, BBQuests::catch_fighting, BBQuests::catch_flying, BBQuests::catch_poison, BBQuests::catch_ground, BBQuests::catch_rock, BBQuests::catch_bug, BBQuests::catch_ghost, BBQuests::catch_steel, 
+    BBQuests::catch_fire, BBQuests::catch_water, BBQuests::catch_grass, BBQuests::catch_electric, BBQuests::catch_psychic, BBQuests::catch_ice, BBQuests::catch_dragon, BBQuests::catch_dark, BBQuests::catch_fairy
+};
+
+//Red quests that appear at the top when a player completes 10 blues.
+//In multiplayer, check if picnic is active before attempting some as only one player can picnic at a time.
+const std::set<BBQuests> red_quests = {
+    BBQuests::wash_pokemon, BBQuests::wild_tera, BBQuests::auto_30, BBQuests::tera_raid, BBQuests::sandwich_three, BBQuests::bitter_sandwich, BBQuests::sweet_sandwich, BBQuests::salty_sandwich, BBQuests::sour_sandwich, BBQuests::spicy_sandwich, BBQuests::hatch_egg, 
+    BBQuests::photo_normal, BBQuests::photo_fighting, BBQuests::photo_flying, BBQuests::photo_poison, BBQuests::photo_ground, BBQuests::photo_rock, BBQuests::photo_bug, BBQuests::photo_ghost, BBQuests::photo_steel, BBQuests::photo_fire, BBQuests::photo_water, 
+    BBQuests::photo_grass, BBQuests::photo_electric, BBQuests::photo_psychic, BBQuests::photo_ice, BBQuests::photo_dragon, BBQuests::photo_dark, BBQuests::photo_fairy
+};
+
+//Multiplayer only. Appears at the top when 3 red quests are completed.
+const std::set<BBQuests> gold_quests = {
+    BBQuests::ditto_central, BBQuests::ditto_canyon, BBQuests::ditto_coastal, BBQuests::ditto_polar, BBQuests::ditto_savanna, BBQuests::group_canyon, BBQuests::group_coastal, BBQuests::group_polar, BBQuests::group_savanna, BBQuests::group_eyewear, BBQuests::group_nonuniform, 
+    BBQuests::group_masks, BBQuests::sandwich_four, BBQuests::catch_hint, BBQuests::catch_hint2
+};
+
+//Quests that are not currently supported. Gold quests currently excluded as this is singleplayer only right now.
+const std::set<BBQuests> not_possible_quests = {
+    BBQuests::UnableToDetect, BBQuests::pickup_10
+};
+
+
+// Open the BBQ panel and read the current amount of BP
+int read_BP(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context);
+
+//Open the quest panel and read in the current quests.
+//Currently only supports singleplayer.
+//For multiplayer, we will want keep track of which quest are gold/red and scroll cursor down until all of the current player's blue quests are in.
+std::vector<BBQuests> read_quests(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context, BBQOption& BBQ_OPTIONS);
```

Comment:
Can `BBQOption& BBQ_OPTIONS` be made const? I don't think you're changing it. But at the same time, I'm don't remember if the getters are const. (since they internally take locks)